src/eva/vision/data/datasets/segmentation/total_segmentator.py

```diff
@@ -134,19 +123,21 @@ def __len__(self) -> int:
         return len(self._indices) * self._n_slices_per_image
 
     @override
-    def load_image(self, index: int) -> np.ndarray:
+    def load_image(self, index: int) -> tv_tensors.Image:
         image_path = self._get_image_path(index)
         slice_index = self._get_sample_slice_index(index)
         image_array = io.read_nifti_slice(image_path, slice_index)
-        return image_array.repeat(3, axis=2)
+        image_rgb_array = image_array.repeat(3, axis=2)
+        return tv_tensors.Image(image_rgb_array.transpose(2, 0, 1))
 
     @override
-    def load_mask(self, index: int) -> np.ndarray:
+    def load_masks(self, index: int) -> tv_tensors.Mask:
         masks_dir = self._get_masks_dir(index)
         slice_index = self._get_sample_slice_index(index)
         mask_paths = (os.path.join(masks_dir, label + ".nii.gz") for label in self.classes)
-        masks = [io.read_nifti_slice(path, slice_index) for path in mask_paths]
-        return np.concatenate(masks, axis=-1)
+        list_of_mask_arrays = [io.read_nifti_slice(path, slice_index) for path in mask_paths]
+        masks = np.concatenate(list_of_mask_arrays, axis=-1)
+        return tv_tensors.Mask(masks.transpose(2, 0, 1))
```

Comment:
I'm not sure if I understand this part 100%. Here we first concatenate on `axis=-1` (this corresponds to the image channel right?) and then we transpose from channel last to channel first format?

Reply:
exactly yes, because the torchvision Mask requires to have channel first

Reply:
to be more specific, why does the concatenation happen along the channel dimension?

Reply:
so each mask - binary mask - is (height, width, 1) which basically represents one label
So here the channel dim is always 1, as the segmentation maps are always binary.

then, to get the multiclass segmentation map, we will need something like (height, width, labels) - I guess that you would probably expecting something like (height, width, 1, labels), but yeah this dim basically is squezed and allows the loss functions to work out-of-the-box CrossEntropyLoss, Dice etc